### PR TITLE
[ADVAPP-528]: Start refactoring assistant

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-528

### Technical Description

Fixes the ability to send a message to a new thread.

### CODE REVIEW

File: app-modules/ai/src/Http/Controllers/RetryMessageController.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    public function __invoke(AiThread $thread, Request $request): JsonResponse
+    {
+        if (! $thread->user()->is(auth()->user())) {
+            abort(404);
+        }
+
+        $content = $request->validate([
+            'content' => ['required', 'max:1000'],
+        ])['content'];
```
Comment (Orrison) [inline app-modules/ai/src/Http/Controllers/RetryMessageController.php:55]: Could we move this validation to a custom request?

File: app-modules/ai/src/Http/Controllers/SendMessageController.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+{
+    public function __invoke(AiThread $thread, Request $request): JsonResponse
+    {
+        if (! $thread->user()->is(auth()->user())) {
+            abort(404);
+        }
+
+        $content = $request->validate([
+            'content' => ['required', 'max:1000'],
+        ])['content'];
```
Comment (Orrison) [inline app-modules/ai/src/Http/Controllers/SendMessageController.php:55]: Same request thought as the retry above.

File: app-modules/ai/src/Http/Controllers/ShowThreadController.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Ai\Models\AiMessage;
+
+class ShowThreadController
+{
+    public function __invoke(AiThread $thread): JsonResponse
+    {
+        if (! $thread->user()->is(auth()->user())) {
+            abort(404);
+        }
```
Comment (Orrison) [inline app-modules/ai/src/Http/Controllers/ShowThreadController.php:50]: Same request though as above

File: app-modules/ai/src/Actions/CreateThread.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        $existingThread = auth()->user()->aiThreads()
+            ->whereNull('name')
+            ->whereBelongsTo($assistant, 'assistant')
+            ->whereDoesntHave('messages')
+            ->first();
+
+        if ($existingThread) {
+            return $existingThread;
+        }
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/ai/src/Actions/CreateThread.php:57]: Yup! So we don't have to worry about cleanup / skewed usage stats :)

File: app-modules/ai/src/Http/Controllers/RetryMessageController.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Ai\Actions\RetryMessage;
+
+class RetryMessageController
+{
+    public function __invoke(AiThread $thread, Request $request): JsonResponse
+    {
+        if (! $thread->user()->is(auth()->user())) {
+            abort(404);
+        }
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/ai/src/Http/Controllers/RetryMessageController.php]: Done

File: app-modules/ai/src/Http/Controllers/RetryMessageController.php (hunk 1 above)
Comment (danharrin) [reply app-modules/ai/src/Http/Controllers/RetryMessageController.php]: Done

File: app-modules/ai/src/Http/Controllers/SendMessageController.php (hunk 2 above)
Comment (danharrin) [reply app-modules/ai/src/Http/Controllers/SendMessageController.php]: Done

File: app-modules/ai/src/Http/Controllers/ShowThreadController.php (hunk 3 above)
Comment (danharrin) [reply app-modules/ai/src/Http/Controllers/ShowThreadController.php]: Done

**Review by Orrison — APPROVED** [review]

**Pull request merged**